pkg/sidecar-controller/snapshot_controller.go

```diff
@@ -67,22 +67,22 @@ func (ctrl *csiSnapshotSideCarController) syncContent(content *crdv1.VolumeSnaps
 			// Note that the deletion snapshot operation will update content SnapshotHandle
 			// to nil upon a successful deletion. At this
 			// point, the finalizer on content should NOT be removed to avoid leaking.
-			err := ctrl.deleteCSISnapshot(content)
+			var err error
+			content, err = ctrl.deleteCSISnapshot(content)
 			if err != nil {
 				return true, err
 			}
-			return false, nil
+			// Continue removing the finalizer
```

Comment:
Is this necessary?  deleteCSISnapshot() will lead to cleanVolumeSnapshotStatus() which will trigger syncContent which will go to the removeContentFinalizer code path.

Reply:
I don't see a reason why not continue in a single sync call. The controller already knows it needs to do it, so why wait? 

Sure, the controller would remove the finalizer on the next sync, and that's what will happen in case the controller crashes.